evalml/tests/automl_tests/test_automl.py

```diff
@@ -5102,6 +5102,25 @@ def test_pipeline_parameter_warnings_only_parameternotused(
     assert str(warning) == "Big bad warning"
 
 
+def test_pipeline_parameter_warnings_with_other_types():
```

Comment:
Is there test coverage on if we get multiple warnings with the same message? ie coverage for
```python
elif str(msg.message) not in raised_messages:
                warnings.warn(msg.message)
                raised_messages.append(str(msg.message))
```
?

Reply:
Yep, this test covers that! The `TargetLeakage` warning actually throws the same warning for each generated pipeline, just like the `ParameterNotUsed` warnings come up. This test is actually how I discovered we needed to filter duplicate warnings.